File: java_solutions/38.count-and-say.java
/*
 * @lc app=leetcode id=38 lang=java
 *
 * [38] Count and Say
 */

// @lc code=start
// semicolons : 1
class Solution { public String countAndSay(int n) { return Stream.iterate("1", s -> java.util.regex.Pattern.compile("(\\d)\\1*").matcher(s).results().map(m -> m.group().length() + m.group().substring(0, 1)).collect(Collectors.joining())) .limit(n) .reduce((a, b) -> b) .orElse("1"); } }
// @lc code=end

File: kotlin_solutions/38.count-and-say.kt
/*
 * @lc app=leetcode id=38 lang=kotlin
 *
 * [38] Count and Say
 */

class Solution {fun countAndSay(n: Int): String = (1 until n).fold("1") { res, _ -> Regex("(.)\\1*").findAll(res).fold("") { acc, m -> acc + m.value.length + m.value[0] } }}

